vite.config.js

```diff
@@ -0,0 +1,62 @@
+import path from 'path';
+import { writeFile } from 'fs/promises';
+import childProcess from 'child_process';
+import pkg from './package.json';
+
+import { defineConfig } from 'vite';
+import { sentryVitePlugin } from '@sentry/vite-plugin';
+import react from '@vitejs/plugin-react';
+
+export default defineConfig(async () => {
+  const isDebug = process.env.NODE_ENV !== 'production';
+  const version = await setupEnv(isDebug);
+
+  let proxy = {};
+  try {
+    proxy = await import('./proxy.config');
+  } catch {}
+
+  let sentryPlugin;
+  if (process.env.SENTRY_AUTH_TOKEN) {
+    sentryPlugin = sentryVitePlugin({
+      authToken: process.env.SENTRY_AUTH_TOKEN,
+      org: 'shoko-anime',
+      project: 'shoko-webui',
+      release: isDebug ? 'dev' : version,
+      include: './dist',
+      urlPrefix: '~/webui/dist/',
+      ignore: [],
+    });
+  }
+
+  return {
+    server: {
+      open: "/webui/",
+      port: 3000,
+      proxy,
+    },
+    resolve: {
+      alias: [
+        { find: '@', replacement: path.resolve(__dirname, 'src') },
+      ],
+    },
+    build: {
+      sourcemap: true,
+    },
+    plugins: [react(), sentryPlugin],
+    base: "/webui/"
+  };
+});
+
+async function setupEnv(isDebug) {
+  const gitHash = childProcess.execSync("git log --pretty=format:'%h' -n 1").toString().replace(/["']/g, '');
+  const appVersion = JSON.stringify(pkg.version);
```

Comment:
Why are we trying to stringify a value that's already a string? I don't think we will ever change it so it's not a string in the `package.json` file, so it's safe to just directly use it as a string here.
```suggestion
 const appVersion = pkg.version;
```

Reply:
was a relic from when I copy/pasted from webpack, I'll remove it.